Refine CSS Value Processing document

sakupi01 opened the pull request:
### Description

The change mainly aims for better accuracy to describe CSS value processing flow according to the specification.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

The previous version sometimes used incorrect terminology and descriptions, which will ultimately lead differences between mdn and the specs.

e.g.

- CSS property value result will not come only from specificity - it will be the result of the entire cascade process
- "specified values" were mostly used incorrectly/confused with "declared values" - "specified values" result from the defaulting process after cascading, but the previous version didn't make this clear
- The document mainly described the stages after the [Defaulting process](https://www.w3.org/TR/css-cascade-4/#defaulting), even though the terminology and phases _before_ Defaulting
 (which are Filtering, Cascading, and related terms) are necessary to make _after_-Defaulting values understandable. Omitting the before-Defaulting process essentially leads to misworded descriptions.
- Initial value shouldn't be listed as a value processing stage according to the spec - it is the terminology used in the specified value resolution process 

The change also includes some typo/wording updates, which will make the description more accurate or simpler.

### Additional details

<!-- 🔗 Link to release notes, browser docs, bug trackers, source control, or other resources. -->

The changes ensure compliance with:
- [CSS Syntax Module Level 3](https://www.w3.org/TR/css-syntax-3/)
- [CSS Cascading and Inheritance Level 4](https://www.w3.org/TR/css-cascade-4/)

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request must be merged first, use "**Depends on:** #123" -->

<!-- 🔎 After submitting, the 'Checks' tab of your PR shows the build status. -->

sakupi01 committed:
renew css value processing document

Signed-off-by: saku <saku@email.sakupi01.com>

Josh-Cena commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+Each property value comes from a single property name-value pair; a single value is applied from each property. Even if the value is a comma separated list of values, that list of values came from a single declaration.
 
-To determine which specified value is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
+To determine which [declared value](#declared-value) is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
```
```suggestion
To determine which [declared value](#declared_value) is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
```

Make sure all your anchor links are valid

Josh-Cena commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
 ## Property values
 
-Every CSS property's value comes from the declaration with the greatest {{cssxref("specificity")}}. If two or more declarations with the same specificity provide different property values for the same element, the declaration value whose selector has the greatest algorithmic weight gets applied.
+Every CSS property's value comes from the declaration that wins the {{cssxref("Cascade")}} algorithm. When multiple declarations provide different property values for the same element, the declaration that ranks highest in the [Cascade Sorting Order](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) gets applied.
```
`CSS/Cascade` is a redirect, plus `{{cssxref("Cascade")}}` would be rendered as code, so you should just use a markdown link instead. Also sentence case.

```suggestion
Every CSS property's value comes from the declaration that wins the [cascade](/en-US/docs/Web/CSS/CSS_cascade/Cascade) algorithm. When multiple declarations provide different property values for the same element, the declaration that ranks highest in the [cascade sorting order](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) gets applied.
```

Same below.

Josh-Cena commented on files/en-us/web/css/css_cascade/value_processing/index.md:
````diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+
+```css
+/* Example of inherited property */
+color: /* inherits from parent's computed value */
````
?

Josh-Cena reviewed (commented):
Drive-by, just looking at Markdown correctness. Not interested in reviewing, although this looks like good work; thank you!

sakupi01 committed:
correct anchor links

sakupi01 committed:
use non-code/redirect ref

sakupi01 committed:
update non-inherited/inherited property code description

sakupi01 commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+Every CSS property's value comes from the declaration that wins the [cascade](/en-US/docs/Web/CSS/CSS_cascade/Cascade) algorithm. When multiple declarations provide different property values for the same element, the declaration that ranks highest in the [cascade sorting order](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) gets applied.
 
-Each property value comes from a single property-value pair; is single value is applied from each property. Even if the value is a comma separated list of values, that list of values came from a single declaration.
+Each property value comes from a single property name-value pair; a single value is applied from each property. Even if the value is a comma separated list of values, that list of values came from a single declaration.
```
property = name + value

see: https://www.w3.org/TR/css-syntax-3/#declaration

estelle replied:
```suggestion
Even when there are multiple [declared values](#declared_value), with multiple declarations providing the same or different property values for the same element, each property value must still come from a single property name-value pair as only a single value is applied from each property. Even if the value is a comma-separated list of values, that list of values came from a single declaration.
```

sakupi01 commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+Each property value comes from a single property name-value pair; a single value is applied from each property. Even if the value is a comma separated list of values, that list of values came from a single declaration.
 
-To determine which specified value is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
+To determine which [declared value](#declared_value) is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
```
The term "specified value" was confusing in this context.
"specified value" comes after declaration value and cascaded value, but here it sounded like it comes before cascading.

https://www.w3.org/TR/css-cascade-4/#value-stages

sakupi01 commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
 ## Property values
 
-Every CSS property's value comes from the declaration with the greatest {{cssxref("specificity")}}. If two or more declarations with the same specificity provide different property values for the same element, the declaration value whose selector has the greatest algorithmic weight gets applied.
+Every CSS property's value comes from the declaration that wins the [cascade](/en-US/docs/Web/CSS/CSS_cascade/Cascade) algorithm. When multiple declarations provide different property values for the same element, the declaration that ranks highest in the [cascade sorting order](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) gets applied.
```
Here, it sounded like property value comes because of only from the resolution of the specificity algorithm, but it's actually not.

sakupi01 commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
-The [cascade](/en-US/docs/Web/CSS/CSS_cascade/Cascade) determines which value should be applied when multiple conflicting styles target the same element. The cascade algorithm defines how user agents combine property values originating from different sources, scopes, and/or layers. When a selector matches an element, the property's [specified value](#specified_value) from the origin with the high ...
+After applying the [cascading](#cascading) rules and defaulting values step by step, the browser ensures the visual presentation matches the processed CSS.
+
+## Processing overview
```
In order not to use "specified value" incorrectly (or to use "declared value" and "cascaded value" correctly), and in order to follow [Value Processing steps](https://www.w3.org/TR/css-cascade-4/#value-stages) based on the spec, the pre-description of three processing steps is essentially needed.

sakupi01 commented:
@estelle 

I would love a review from you, if that's possible!

estelle commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
 ## Property values
 
-Every CSS property's value comes from the declaration with the greatest {{cssxref("specificity")}}. If two or more declarations with the same specificity provide different property values for the same element, the declaration value whose selector has the greatest algorithmic weight gets applied.
+Every CSS property's value comes from the declaration that wins the [cascade](/en-US/docs/Web/CSS/CSS_cascade/Cascade) algorithm. When multiple declarations provide different property values for the same element, the declaration that ranks highest in the [cascade sorting order](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) gets applied.
```
```suggestion
Every CSS property is set to its [cascaded value](#cascaded_value), which comes from the property declaration that ranks highest in the [cascade sorting order](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) based on the [cascade algorithm](/en-US/docs/Web/CSS/CSS_cascade/Cascade).
```
If the term is cascaded value, let's use that. And leave the "win" to that area and move the "multiple declarations" to the next paragraph.

sakupi01 replied:
> Every CSS property is set to its [cascaded value](#cascaded_value),

Every CSS property doesn't necessarily have its cascaded value - if there are multiple declarations set, there's a cascaded value. 
see: https://www.w3.org/TR/css-cascade-4/#stages-examples

So, I mended the phrase a bit, embracing your suggestion!

sakupi01 replied:
fixed in: [modify&split the first paragraph](https://github.com/mdn/content/pull/40546/commits/118b89e3c623f62af28c98e557f32c3b437fecf3)

estelle commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
-To determine which specified value is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
+To determine which [declared value](#declared_value) is applied, the user agent gathers and processes all the styles from different sources, such as inline styles, and internal and external stylesheets.
+
+The [cascade](/en-US/docs/Web/CSS/CSS_cascade/Cascade) determines which value should be applied when multiple conflicting styles target the same element. The [cascade algorithm](/en-US/docs/Web/CSS/CSS_cascade/Cascade#cascading_order) defines how user agents combine property values originating from different sources, scopes, and/or layers. When a selector matches an element, the property's [decla ...
```
let's link the terms "layer" and "origin"

sakupi01 replied:
fixed in: [links for layers and origins](https://github.com/mdn/content/pull/40546/commits/57d9f3511c4cfb4ff82cff0af87e6f04fc5d3e33)

estelle commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+**Filtering** is the process of identifying all declarations that apply to each element. During this phase, the browser:
+
+- Verifies that declarations belong to active stylesheets
+- Checks that property names are valid and recognized
```
can we merge #2 into #4? If the property name is not valid or recognized, it's invalid according to the browser (even if valid in supporting browsers. 

We may also want to link to the error processing (https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_syntax/Error_handling) and link from there back to here.

estelle commented on files/en-us/web/css/css_cascade/value_processing/index.md:
````diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+}
+```
+
+As a result of filtering, each element has zero or more [**declared values**](#declared_value) for each property. These declared values are the starting point for the next processing stages; [Cascading](#cascading).
````
```suggestion
As a result of filtering, each element has zero or more [**declared values**](#declared_value) for each property. These declared values are the starting point for the next processing stages; [cascading](#cascading).
```

estelle commented on files/en-us/web/css/css_cascade/value_processing/index.md:
````diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+- Eliminates declarations that are syntactically invalid
+
+```css
+/* Valid declarations become declared values */
+p {
+  font-size: 1.25em; /* Valid */
+  colr: blue; /* Invalid property name - filtered out */
+  font-weight: bold; /* Valid */
````
```suggestion
p {
  font-size: 1.25em;
  colr: blue;
  font-weight: bold;
```

estelle reviewed (commented):
This PR is excellent. I only got through the first bit, so have more to review, but I am heading out in 20 minutes for the week so I wanted to commit what I have instead of leaving you waiting.

I realize from the number of comments in just the first portion of this it seems like i may be being critical. I am just trying to ensure that the level is such that people with less knowledge than us aren't confused. So please don't take the number of edits as a reflection of the work you did. It's excellent. I am just making suggestions that I hope will answer the readers questions before they ask them.

sakupi01 committed:
Apply suggestion from @estelle

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sakupi01 committed:
Apply suggestion from @estelle

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sakupi01 committed:
Apply suggestion from @estelle

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sakupi01 committed:
Apply suggestion from @estelle

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

oihdsf committed:
links for layers and origins

oihdsf committed:
modify&split the first paragraph

oihdsf committed:
make all code comments removed and replaced with prose explanations

oihdsf committed:
lowercased the link texts

oihdsf committed:
remove spaces

oihdsf committed:
merge bullet items and add link to error handling

oihdsf committed:
filtering: include suggestion

sakupi01 commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,156 @@ For every element in a document tree, the browser assigns a value to every CSS p
+- Checks that property names are valid and recognized
+- Validates that property values match the expected syntax
+- Eliminates declarations that are syntactically invalid
+
```
fixed in: [make all code comments removed and replaced with prose explanations](https://github.com/mdn/content/pull/40546/commits/10974c491fc927dedbae4ab871a2c400d3955fe6)

oihdsf committed:
Revert "merge bullet items and add link to error handling"

This reverts commit 49ed161bcdc9ea6f362c993fee04612f5f397bc7.

oihdsf committed:
merge bullet items and add link to error handling

sakupi01 commented:
Thank you so much for taking the time to review this, @estelle! 

I really appreciate your thoughtful feedback, especially knowing how busy you are.
I've addressed the points you raised and commented respectively. 
Looking forward to your thoughts on the rest when you get a chance. 

Enjoy your time off!

estelle reviewed (commented):
a few more suggestions and questions

sakupi01 committed:
Update files/en-us/web/css/css_cascade/value_processing/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sakupi01 committed:
Update files/en-us/web/css/css_cascade/value_processing/index.md

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

sakupi01 commented on files/en-us/web/css/css_cascade/value_processing/index.md:
```diff
@@ -10,50 +10,152 @@ For every element in a document tree, the browser assigns a value to every CSS p
+
+**Defaulting** ensures every property on every element has a value, even when no declarations apply. This involves:
+
+- Using **inherited values** for [inherited properties](/en-US/docs/Web/CSS/CSS_cascade/Inheritance#inherited_properties)
+- Using **initial values** for [non-inherited properties](/en-US/docs/Web/CSS/CSS_cascade/Inheritance#non-inherited_properties)
+- Applying **explicit defaulting keywords** ([`initial`](/en-US/docs/Web/CSS/initial), [`inherit`](/en-US/docs/Web/CSS/inherit), [`unset`](/en-US/docs/Web/CSS/unset), [`revert`](/en-US/docs/Web/CSS/revert), [`revert-layer`](/en-US/docs/Web/CSS/revert-layer))
```
`inherit` and `initial` are only ones that are explicitly set by parser, as you mentioned.
However, explicit defaulting keywords also contributes to set specified values which is a result of Defaulting process here. (Each explicit defaulting keywords are also converted to proper specified values.)
So, I thought it's better to mention about explicit defaulting keywords conversion, not entirely deleting.
What do you think about: ?

oihdsf committed:
all grammatical changes

oihdsf committed:
fixed defaulting description with accuracy

sakupi01 committed:
Add 'Actual value' to CSS value processing index

estelle reviewed (approved):
Thank you!

estelle reviewed (approved):
Thank you! 
Yay. That was quite the first content PR. Welcome to the team!

estelle committed:
Merge branch 'main' into fix/value-processing
